test/Microsoft.NET.Build.Tests/DesignTimeBuildTests.cs

```diff
@@ -108,6 +108,8 @@ public void DesignTimePackageDependenciesAreResolved(string targetFramework)
 
             testProject.PackageReferences.Add(new TestPackageReference("Newtonsoft.Json", ToolsetInfo.GetNewtonsoftJsonPackageVersion(), privateAssets: "All"));
             testProject.PackageReferences.Add(new TestPackageReference("Humanizer", "2.8.26"));
+            // disable implicit use of the Roslyn Toolset compiler package
+            testProject.AdditionalProperties["BuildWithNetFrameworkHostedCompiler"] = false.ToString();
```

Comment:
What happens when this is not set in the tests where you set it?

Reply:
The test assertions are now wrong because they're asserting things about the PackageReferences in the project.

Reply:
For clarity - in some cases there's a new implicit packagereference, and instead of adding that reference to all of these locations I figured it made more sense to control the addition of the implicit reference in the first place more explicitly.